ui/analyse/src/study/studyCtrl.ts

```diff
@@ -491,8 +492,8 @@ export default class StudyCtrl {
       this.vm.mode.sticky = false;
       if (!this.vm.behind) this.vm.behind = 1;
       this.vm.chapterId = id;
+      componentCallbacks(id);
       await this.xhrReload();
-      componentCallbacks();
```

Comment:
moving `componentCallbacks()` before `xhrReload()` is likely to break something. Is it necessary, why?

Reply:
It allows the redraw of the live board to happen at the same time as the main board. otherwise that redraw lags behind a second for some UI flows as it needs onChapterChange. we could insert extra redraw logic to keep the two in sync for all of the various ways in which a chapter can be selected, but i went with MultiRedraw and onChapterChange to cover all the bases.

What could it break?

Reply:
Well, `componentCallbacks` is supposed to tell components about things having changed. It used to do it **after** things have changed, and now it does it before, so it seemed like it could break something.
But if it doesn't :shrug: